src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,878 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
+                var analyzer = analyzerAssembly.CreateInstance("Analyzer")!;
+                analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
+                Assert.Equal(ExecutionConditionUtil.IsCoreClr ? "1" : "42", sb.ToString());
+            });
         }
 
-        [ConditionalFact(typeof(WindowsOnly), typeof(CoreClrOnly))]
-        public void AssemblyLoading_NativeDependency()
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/66104")]
+        public void AssemblyLoading_CompilerDependencyDuplicated()
         {
-            var loader = new DefaultAnalyzerAssemblyLoader();
-            loader.AddDependencyLocation(_testFixture.AnalyzerWithNativeDependency.Path);
+            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            {
+                var assembly = typeof(ImmutableArray<int>).Assembly;
+
+                // Copy the dependency to a new location to simulate it being deployed by an 
+                // analyzer / generator
+                using var tempRoot = new TempRoot();
+                var destFile = tempRoot.CreateDirectory().CreateOrOpenFile($"{assembly.GetName().Name}.dll");
+                destFile.CopyContentFrom(assembly.Location);
+                loader.AddDependencyLocation(destFile.Path);
+
+                var copiedAssembly = loader.LoadFromPath(destFile.Path);
+                Assert.Single(AppDomain.CurrentDomain.GetAssemblies().Where(x => x.FullName == assembly.FullName));
+                Assert.Same(copiedAssembly, assembly);
+            });
+        }
 
-            Assembly analyzerAssembly = loader.LoadFromPath(_testFixture.AnalyzerWithNativeDependency.Path);
-            var analyzer = analyzerAssembly.CreateInstance("Class1")!;
-            var result = analyzer.GetType().GetMethod("GetFileAttributes")!.Invoke(analyzer, new[] { _testFixture.AnalyzerWithNativeDependency.Path });
-            Assert.Equal(0, Marshal.GetLastWin32Error());
-            Assert.Equal(FileAttributes.Archive, (FileAttributes)result!);
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void AssemblyLoading_NativeDependency()
+        {
+            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            {
+                const int INVALID_FILE_ATTRIBUTES = -1;
+                loader.AddDependencyLocation(testFixture.AnalyzerWithNativeDependency.Path);
+
+                Assembly analyzerAssembly = loader.LoadFromPath(testFixture.AnalyzerWithNativeDependency.Path);
+                var analyzer = analyzerAssembly.CreateInstance("Class1")!;
+                var result = analyzer.GetType().GetMethod("GetFileAttributes")!.Invoke(analyzer, new[] { testFixture.AnalyzerWithNativeDependency.Path });
+                Assert.NotEqual(INVALID_FILE_ATTRIBUTES, result);
```

Comment:
Can you explain what changed here?

Reply:
This change cost me about one hour of my life to track down. The real issue is that `GetLastWin32Error` is only interesting to call when a native interop call fails. When that call succeeds it has unknown data inside of it. That is because a native call only sets the last error on failure, on success it does nothing. 

In the change where I fix the bug I added a `File.Exists` check on this code path. That check fails and sets the last error to 2. No other native call fails on this code path so when we get to this call the `Assert.Equal` failed because of that prior `File.Exists` failure. 

The actual failure code for `GetFileAttributesW` is `INVALID_FILE_ATTRIBUTES` so I moved the check to be that. Really I considered just deleting this entirely as the `Assert.Equal` check below is sufficient to verify this case.